feat(docs): Better docs for big chunk of schema stuff

freiksenet opened the pull request:
New docs for big chunk of schema stuff. sift docs need to be fully rewritten / changed to node-model docs. Inference stuff will be re-written by Vlad.

<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)
-->

<!--
  Is this a blog post? Check out the docs at https://www.gatsbyjs.org/contributing/blog-contributions/, and please mention if the blog post is pre-approved
  by someone from Gatsby.
-->

## Description

<!-- Write a brief description of the changes introduced by this PR -->

### Documentation

<!--
  Where is this feature or API documented?

  - If docs exist:
    - Update any references, if relevant. This includes Guides and Gatsby Internals docs.
  - If no docs exist:
    - Create a stub for documentation including bullet points for how to use the feature, code snippets (including from happy path tests), etc.
  - Tag @gatsbyjs/learning for review, pairing, polishing of the documentation
-->

## Related Issues

<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234

  Link to an issue that is partially addressed by this PR (if there are any)
  e.g. Addresses #1234

  Link to related issues (if there are any)
  e.g. Related to #1234
-->

yg commented on docs/docs/schema-generation.md:
```diff
@@ -2,66 +2,68 @@
 
-> This documentation isn't up to date with the latest [schema customization changes](/docs/schema-customization).
-> You can help by making a PR to [update this documentation](https://github.com/gatsbyjs/gatsby/issues/14228).
+Once the nodes have been sourced and transformed, the next step is to generate the GraphQL Schema. Gatsby Schema is different from many regular GraphQL schemas is that it's combines plugin or user defined schema information with data inferred from node shapes itself. The latter is called _schema inferrence_. Users or plugin can explicity define whole or part of the schema using [schema customizat ...
```
Fix Grammar and Typo

```suggestion
Once the nodes have been sourced and transformed, the next step is to generate the GraphQL Schema. Gatsby Schema is different from many regular GraphQL schemas is that it combines plugin or user defined schema information with data inferred from node shapes itself. The latter is called _schema inferrence_. Users or plugin can explicitly define whole or part of the schema using [schema customization API](/docs/schema-customization). Usually every node will get a GraphQL Type based on its `node.internal.type` field. When using Schema Customization, all types that implement `Node` interface become GraphQL Node Types and thus get root level fields for accessing them.
```

yg commented on docs/docs/schema-generation.md:
```diff
@@ -2,66 +2,68 @@
+### `graphql-config.js` mapping
 
-This step creates GraphQL input filters for each field so the objects can be queried by them. More details in [Building the Input Filters](/docs/schema-input-gql).
+It's possible to connect types throught site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
```
Typo

```suggestion
It's possible to connect types throughout site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
```

yg commented on docs/docs/schema-generation.md:
```diff
@@ -2,66 +2,68 @@
+
+## 6. Merging in third-party schemas
+
+If a plugin like `gatsby-source-graphql` is used, all third-party schemas that it provided are merged in into the Gatsby schema.
```
Remove preposition

```suggestion
If a plugin like `gatsby-source-graphql` is used, all third-party schemas that it provided are merged into the Gatsby schema.
```

yg commented on docs/docs/schema-root-fields.md:
```diff
@@ -0,0 +1,106 @@
+
+## Singular root fields
+
+Singular root fields accept same `filter` parameter as plural, but the `filter` is spread directily into arguments. Thus filter parameters need to be passed to the field directly. They return the resulting object directly.
```
Typo

```suggestion
Singular root fields accept same `filter` parameter as plural, but the `filter` is spread directly into arguments. Thus filter parameters need to be passed to the field directly. They return the resulting object directly.
```

freiksenet commented:
@AishaBlake Does removing the label means it's good to merge?

AishaBlake commented:
Really sorry for the delay, @freiksenet! Removing the label just means it's been triaged.

freiksenet commented on docs/docs/schema-generation.md:
```diff
@@ -2,66 +2,68 @@
+It's possible to connect types throughout site configuration using [Node Type Mapping](https://www.gatsbyjs.org/docs/gatsby-config/#mapping-node-types).
 
-### 4. ProcessedTypeNode creation with resolve implementation
+It is a convenient escape hatch and there are no plans to deprecate this API at the moment.
```
Fixed the phrasing.

freiksenet commented:
@AishaBlake Thanks a lot for the review! I've fixed/accepted issues that you've raised, added redirects and rebased. It's now ready for re-review and a merge.

AishaBlake commented:
Thanks, @freiksenet! I see the linter is picking up on [a few things](https://circleci.com/gh/gatsbyjs/gatsby/476732?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). If you get to this before me, would you mind fixing those? If not, I'll take care of them. This is one of the first things on my "to review" list for next week.

freiksenet commented:
@AishaBlake Fixed the linter issues now :)
